elements/howto-checkbox/howto-checkbox.js

```diff
@@ -0,0 +1,161 @@
+   */
+  const KEYCODE = {
+    SPACE: 32,
+  };
+
+  /**
+   * The `HowToCheckbox` exposes a single `checked` attribute/property for
+   * toggling its state. Changes to the `checked` property will also be
+   * reflected to an `aria-checked` attribute. Similarly, the `disabled`
+   * property is reflected to an `aria-disabled` attribute. This controls
+   * whether the element is operable or not.
+   */
+  class HowToCheckbox extends HTMLElement {
+    static get observedAttributes() {
+      return ['checked', 'disabled'];
+    }
+
+    /**
+     * `connectedCallback` sets the initial `role`, `tabindex`,
+     * internal state, and installs event listeners.
+     */
+    connectedCallback() {
+      if (!this.hasAttribute('role'))
+        this.setAttribute('role', 'checkbox');
+      if (!this.hasAttribute('tabindex'))
+        this.setAttribute('tabindex', 0);
+
+      this._checked = false;
+      this._disabled = false;
+
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('click', this._onClick);
+    }
+
+    _onKeyDown(event) {
+      // Don’t handle modifier shortcuts typically used by assistive technology.
+      if (event.altKey)
+        return;
+
+      switch (event.keyCode) {
+        case KEYCODE.SPACE:
+          this._toggleChecked();
+          break;
+        // Any other key press is ignored and passed back to the browser.
+        default:
+          return;
+      }
+    }
+
+    _onClick(event) {
+      this._toggleChecked();
+    }
+
+    /**
+     * `_toggleChecked` calls the checked setter and flips its state.
+     * Because `_toggleChecked` is only caused by a user action, it will
+     * also dispatch a change event.
+     */
+    _toggleChecked() {
+      if (this.disabled)
+        return;
+      this.checked = !this.checked;
+      this.dispatchEvent(new CustomEvent('change', {
+        detail: {
+          checked: this.checked,
+        },
+        bubbles: false,
+      }));
+    }
+
+    /**
+     * Because setting either the `checked` or `disabled` property also
+     * sets the corresponding attributes, it's possible to get into a cycle
+     * where setting a property sets the attribute which then tries to
+     * set the property again. This helper avoids that by checking to see
+     * if the attribute is already set. As an additional helper, it will
+     * also use "true"/"false" strings for boolean ARIA attributes.
+     */
+    _toggleAttribute(attr, value) {
+      const isAriaAttribute = attr.indexOf('aria-') === 0;
+      if (value === true && !this.hasAttribute(attr)) {
+        if (isAriaAttribute)
+          this.setAttribute(attr, 'true');
+        else
+          this.setAttribute(attr, '');
+        return;
+      }
+      if (value === false && this.hasAttribute(attr)) {
+        if (isAriaAttribute)
+          this.setAttribute(attr, 'false');
+        else
+          this.removeAttribute(attr);
+        return;
+      }
+    }
```

Comment:
I think this can be simplified:

```js
const isAriaAttribute = attr.indexOf('aria-') === 0;
if (isAriaAttribute) {
  // aria-* attributes need strings "true" or "false" as values
  this.setAttribute(attr, value.toString());
  return;
}
if (value === false) this.removeAttribute(attr);
if (value === true && !this.hasAttribute(attr)) this.setAttribute(attr, '');
```